pkg/icingadb/ha.go

```diff
@@ -259,11 +259,11 @@ func (h *HA) realize(s *icingaredisv1.IcingaStatus, t *types.UnixMilli, shouldLo
 	return nil
 }
 
-func (h *HA) removeInstance() {
+func (h *HA) removeInstance(ctx context.Context) {
 	h.logger.Debugw("Removing our row from icingadb_instance", zap.String("instance_id", hex.EncodeToString(h.instanceId)))
 	// Intentionally not using a context here as this is a cleanup task and h.ctx is already cancelled.
```

Comment:
That comment is now outdated as some context is used now. IMHO it would also be fine to just create the context with timeout right here.